# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webcodecs/VideoDecoder.h
# Commit: 6c61c412fe0b
# Full Hash: 6c61c412fe0b1afbec5d58dd6fea7c76d0531ff6
# Author: Paul Adenot <paul@paul.cx>
# Date: 2023-06-30 04:06:43
# Regressor Bug: 1831451
# File Overlap Count: 1
# Description:
#   Bug 1831451 - Don't use DOM objects internally in Web Codecs implementation. r=chunmin
#   
#   Depends on D181328
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D182476
# ==============================================================================

diff -r 7dd351fce5ad -r 6c61c412fe0b dom/media/webcodecs/VideoDecoder.h
--- a/dom/media/webcodecs/VideoDecoder.h	Thu Jun 29 22:33:14 2023 +0000
+++ b/dom/media/webcodecs/VideoDecoder.h	Thu Jun 29 22:33:15 2023 +0000
@@ -19,9 +19,12 @@
 #include "mozilla/Result.h"
 #include "mozilla/UniquePtr.h"
 #include "mozilla/dom/BindingDeclarations.h"
+#include "mozilla/dom/RootedDictionary.h"
+#include "mozilla/dom/VideoColorSpaceBinding.h"
 #include "nsCycleCollectionParticipant.h"
 #include "nsStringFwd.h"
 #include "nsWrapperCache.h"
+#include "mozilla/dom/VideoDecoderBinding.h"
 
 class nsIGlobalObject;
 
@@ -54,6 +57,29 @@
 
 namespace mozilla::dom {
 
+struct VideoColorSpaceInternal {
+  explicit VideoColorSpaceInternal(const VideoColorSpaceInit& aColorSpaceInit);
+  VideoColorSpaceInternal() = default;
+  VideoColorSpaceInit ToColorSpaceInit() const;
+  Maybe<bool> mFullRange;
+  Maybe<VideoMatrixCoefficients> mMatrix;
+  Maybe<VideoColorPrimaries> mPrimaries;
+  Maybe<VideoTransferCharacteristics> mTransfer;
+};
+
+struct VideoDecoderConfigInternal {
+  explicit VideoDecoderConfigInternal(const VideoDecoderConfig& aConfig);
+  nsString mCodec;
+  Maybe<uint32_t> mCodedHeight;
+  Maybe<uint32_t> mCodedWidth;
+  Maybe<VideoColorSpaceInternal> mColorSpace;
+  Maybe<RefPtr<MediaByteBuffer>> mDescription;
+  Maybe<uint32_t> mDisplayAspectHeight;
+  Maybe<uint32_t> mDisplayAspectWidth;
+  HardwareAcceleration mHardwareAcceleration;
+  Maybe<bool> mOptimizeForLatency;
+};
+
 class ConfigureMessage;
 class DecodeMessage;
 class FlushMessage;
@@ -155,7 +181,7 @@
 
   // Returns true when mAgent can be created.
   bool CreateDecoderAgent(DecoderAgent::Id aId,
-                          UniquePtr<VideoDecoderConfig>&& aConfig,
+                          UniquePtr<VideoDecoderConfigInternal>&& aConfig,
                           UniquePtr<TrackInfo>&& aInfo);
   void DestroyDecoderAgentIfAny();
 
@@ -174,7 +200,7 @@
   // will be destroyed when "reset" or another "configure" is called (spec
   // allows calling two "configure" without a "reset" in between).
   RefPtr<DecoderAgent> mAgent;
-  UniquePtr<VideoDecoderConfig> mActiveConfig;
+  UniquePtr<VideoDecoderConfigInternal> mActiveConfig;
   uint32_t mDecodeQueueSize;
   bool mDequeueEventScheduled;
 